### DISCUSSION TITLE
Issue with Product Name Searches Using IndexService in Ecommerce Framework

### DISCUSSION THREAD

**Discussion by AlephLau** [pimcore/pimcore#16899] [Q&A / Support]
Hello Pimcore Team,

I'm encountering a peculiar issue with product searches within Pimcore's Ecommerce Framework, specifically when using IndexService in conjunction with fulltext searches. I've observed that when product names follow specific formats like "IN-25", "xa-25", "XP-12", "l-01", the search fails to locate these products. Interestingly, if I slightly alter the names by adding an extra letter (e.g., "INA-25" or "xaa-25"), the searches then successfully find the products.

I'm utilizing the ecommerceframework_optimized_productindex table for these operations, and my code includes the line $productListing->addQueryCondition($term); to apply search conditions.

I'm looking to understand the behavior of the addQueryCondition method better to delve deeper into this issue and find a solution. So far, I've only found the interface in Pimcore\Bundle\EcommerceFrameworkBundle\IndexService\ProductList, but I'd like to know where I can find the actual implementation of this method or how I might debug this behavior.

Do you have any suggestions on why this issue might be occurring or where I could look to resolve it? Any guidance or recommendations would be greatly appreciated.

Thank you in advance for your assistance.

**Comment by AlephLau** [comment]
@fabfischer @mattamon  Could you please help me with this?

**Comment by markus-moser** [comment]
@AlephCode I guess you are using the default full text search mechanism right? This at the end leads to a MySQL full text `match against` query:

https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/IndexService/ProductList/DefaultMysql.php#L546
https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/IndexService/ProductList/DefaultMysql/Dao.php#L290

And this has some special behaviours like you described it.

To change this you could overwrite the productlist but to be honest I never really used this default full text search conditions in my projects as most of the time there are some special project specific requirements. Therefore I normally just used `addCondition` (https://github.com/pimcore/ecommerce-framework-bundle/blob/1.x/src/IndexService/ProductList/DefaultMysql.php#L91) to add my own customized conditions (most of the time either some `like` conditions or a combination of `like` and `match against`).
